papas/pfalgo/pfreconstructor.py

```diff
@@ -409,9 +410,9 @@ def reconstruct_track(self, track, pdgid, parent_ids,
         mass, charge = particle_data[pdgid]
         p4 = TLorentzVector()
         p4.SetVectM(track.p3, mass)
-        particle = Particle(p4, vertex, charge, pdgid, subtype='r')
+        particle = Particle(p4, vertex, charge, len(self.particles), pdgid,  subtype='r')
         #todo fix this so it picks up smeared track points (need to propagagte smeared track)
-        particle.set_path(track.path)
+        particle.set_track(track) #refer to existing track rather than make a new one
```

Comment:
I can't remember all the details but I made changes here as otherwise the code was creating a new track and this is not needed (I think) and created a discrepancy with cpp. There was also another issue where the track and smeared track got a bit muddled. A rethink about particles, paths and tracks is probably needed.